From e63aa58f82dde6303122c5db16f5b26e319d5134 Mon Sep 17 00:00:00 2001
From: "suryaprakash.konduru" <suryaprakash.konduru@nxp.com>
Date: Tue, 17 Sep 2024 14:50:48 +0530
Subject: [PATCH 11/18] [artf1178427] Allowing only needed polling loop
 information

- Allowing only Google polling notification relevant data and drop
unwanted/extra information

Change-Id: I61dad4230cc2efb7c988dac74e8aa2a30de962ed
---
 .../observe_mode/ReaderPollConfigParser.cc    | 22 ++++---------------
 1 file changed, 4 insertions(+), 18 deletions(-)

diff --git a/snxxx/halimpl/observe_mode/ReaderPollConfigParser.cc b/snxxx/halimpl/observe_mode/ReaderPollConfigParser.cc
index b9bf16c4..08824a52 100644
--- a/snxxx/halimpl/observe_mode/ReaderPollConfigParser.cc
+++ b/snxxx/halimpl/observe_mode/ReaderPollConfigParser.cc
@@ -206,10 +206,7 @@ vector<uint8_t> ReaderPollConfigParser::getEvent(vector<uint8_t> p_event,
             break;
 
           default:
-            event_data = getUnknownEvent(
-                vector<uint8_t>(p_event.begin() + INDEX_OF_L2_EVT_TYPE,
-                                p_event.end()),
-                std::move(timestamp), lastKnownGain);
+            break;
         }
         break;
 
@@ -223,10 +220,6 @@ vector<uint8_t> ReaderPollConfigParser::getEvent(vector<uint8_t> p_event,
         break;
 
       default:
-        event_data = getUnknownEvent(
-            vector<uint8_t>(p_event.begin() + INDEX_OF_L2_EVT_TYPE,
-                            p_event.end()),
-            std::move(timestamp), lastKnownGain);
         break;
     }
 
@@ -256,10 +249,7 @@ vector<uint8_t> ReaderPollConfigParser::getEvent(vector<uint8_t> p_event,
             }
             break;
           default:
-            event_data = getUnknownEvent(
-                vector<uint8_t>(p_event.begin() + INDEX_OF_CMA_EVT_DATA,
-                                p_event.end()),
-                std::move(timestamp), lastKnownGain);
+            break;
         }
         break;
       case CMA_DATA_TRIGGER_TYPE: {
@@ -268,12 +258,8 @@ vector<uint8_t> ReaderPollConfigParser::getEvent(vector<uint8_t> p_event,
         unknownEventTimeStamp = timestamp;
         break;
       }
-      default: {
-        vector<uint8_t> payloadData = vector<uint8_t>(
-            p_event.begin() + INDEX_OF_CMA_EVT_TYPE, p_event.end());
-        event_data = getUnknownEvent(std::move(payloadData),
-                                     std::move(timestamp), lastKnownGain);
-      }
+      default:
+        break;
     }
   } else if (cmaEventType == CMA_EVT_EXTRA_DATA_TAG &&
              readExtraBytesForUnknownEvent) {
-- 
2.47.0

